File: StockMarcketCharting/StockMarketCharting/src/main/java/com/khushboo/StockMarketCharting/service/StockExchangeService.java
package com.khushboo.StockMarketCharting.service;

import com.khushboo.StockMarketCharting.dto.CompanyDto;
import com.khushboo.StockMarketCharting.dto.StockExchangeDto;
import com.khushboo.StockMarketCharting.model.Company;

import java.util.List;

public interface StockExchangeService {
    public List<StockExchangeDto> findAll();
    public StockExchangeDto findById(int id);
    public StockExchangeDto save(StockExchangeDto stockExchangeDto);
    public StockExchangeDto update(StockExchangeDto stockExchangeDto);
    public void delete(int id);
    public List<CompanyDto> getCompanies(int id);
    public StockExchangeDto addCompanyToStockExchange(String stockExchangeName,String companyCode, Company company);

}

File: New folder (4)/frontend/services/stockexchange.service.js
import http from "../http-common";

class StockExchangeService {

  findAll() {
    return http.get(`/stockExchanges`);
  }

  findById(id) {
    return http.get(`/stockExchanges/${id}`);
  }

  create(data) {
    return http.post(`/stockExchanges`, data);
  }

  update(data) {
    return http.put(`/stockExchanges`, data);
  }

  delete(id) {
    return http.delete(`/stockExchanges/${id}`);
  }

  findCompanies(id) {
    return http.get(`/stockExchangescomp/${id}`);
  }
}

export default new StockExchangeService();